### ISSUE TITLE
Examples on Multivariate Time Series

### ISSUE THREAD

**Issue by kayuksel** [torchgan/torchgan#90]
The current framework is too much oriented towards Computer Vision problems where there is already lots of available resources. It would be great if you can also focus on multivariate time series where Generative Adversarial Networks can have several very important uses in data generation, augmentation, imputation, denoising and detecting anomalies for several applications such as medical, financial, etc.. It would be great if you can add examples for few of these use-cases. Enclosed is such dataset consisting of log-returns where cubic interpolation is employed for imputation and market with an 'Imputated' boolean, I am sending it in case you would like to make an example with such a common use-case in financial world.

[Dataset.zip](https://github.com/torchgan/torchgan/files/2948358/Dataset.zip)

**Comment by avik-pal** [comment]
This seems like a really interesting area. We have been planning to diversify this framework and this looks like an interesting direction to proceed. I could find the following papers on this topic. (Feel free to extend the list):
- [ ] http://proceedings.mlr.press/v80/che18a/che18a.pdf
- [ ] https://arxiv.org/pdf/1809.04758.pdf
- [ ] https://arxiv.org/pdf/1901.04997.pdf

**Comment by kayuksel** [comment]
- http://papers.nips.cc/paper/7432-multivariate-time-series-imputation-with-generative-adversarial-networks.pdf

**Comment by kayuksel** [comment]
For the weekend imputation problem, one can create a training set that consists of data from Monday to Friday at each week. Then, impute the weekend before or after using the Generator. Lastly, the imputed imputed seven-days signal can be fed to Discriminator, so that predicts whether the first-two days or last two-days of that signal was imputed. Note: One can probably impute both prior and after weekend at the same time and take the average of the loss of the prediction for each case on each sample week.

**Comment by kayuksel** [comment]
I have attempted making an implementation of what I have described as a potential solution to the weekend imputation problem. I would be more than glad if you can check the implementation; as I am quite familiar with GANs and this was my initial attempt to employ them in a problem:
[1d_gan.zip](https://github.com/torchgan/torchgan/files/2984705/1d_gan.zip)

**Comment by avik-pal** [comment]
@kayuksel Thanks for the attempt. However, we would really appreciate if you could put the code in some git repository or maybe a gist. It makes reviewing it much simpler.

**Comment by avik-pal** [comment]
BTW I did go through your code. It seems fine. The only issue is that it is not exactly the style we follow in torchgan, you could actually use the convenience functions that we provide instead of writing everything explicitly. That being said it is not very difficult to modify it. I would recommend that you open a PR with this in the example directory as a script for now. I can then review it and suggest the changes that you can make.

Overall it seems like a good demonstration for a non-CV Problems. @Aniket1998  and I have had discussions regarding expanding the base of this framework to support a wide variety of GANs and this seems to be a good start.
